photon-api/src/main/scala/com/linkedin/photon/ml/util/DefaultIndexMapLoader.scala

```diff
@@ -19,27 +19,23 @@ import org.apache.spark.broadcast.Broadcast
 
 /**
  * A loader that provides instances of DefaultIndexMap
- *
- * NOTE: instances of this class cannot be used before "prepare" has been called with a SparkContext!
  */
-class DefaultIndexMapLoader(@transient val featureNameToIdMap: Map[String, Int]) extends IndexMapLoader {
+class DefaultIndexMapLoader(sc: SparkContext, featureNameToIdMap: Map[String, Int]) extends IndexMapLoader {
+
   @transient
-  private var _indexMap: IndexMap = _
-  private var _mapBroadCaster: Broadcast[Map[String, Int]] = _
+  private val _indexMap: IndexMap = new DefaultIndexMap(featureNameToIdMap)
```

Comment:
I vote no for private-underscores because underscore is a very prevalent feature of Scala, and including it in var/val names could cause visual confusion.